grammars/gfm.cson

```diff
@@ -867,6 +867,30 @@
         'name': 'markup.underline.link.gfm'
   }
   {
+    'begin': '^\\s*(- \\[ \\])(\\s*)(.*)'
+    'end': '$'
+    'name': 'task.unordered.list.gfm'
+    'captures':
+      '1':
+        'name': 'punctuation.gfm'
+      '2':
+        'name': 'space.gfm'
+      '3':
+        'name': 'text.gfm'
+  }
+  {
+    'begin': '^\\s*(- \\[x\\])(\\s*)(.*)'
+    'end': '$'
+    'name': 'completed.task.unordered.list.gfm'
+    'captures':
+      '1':
+        'name': 'punctuation.gfm'
+      '2':
+        'name': 'space.gfm'
+      '3':
+        'name': 'text.gfm'
```

Comment:
You need the following here if you want to parse markdown on the line itself:

```coffee
'patterns': [
      {
        'include': '$self'
      }
    ]
```

This causes it to use all the existing patterns between the start and end patterns to further tokenize it.